From aed108893bee6bb585f31b21fb4a02ec5c9ca547 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Fri, 21 Feb 2014 10:50:47 +0900
Subject: [PATCH 10/24] waylandsink: Ensure that the frame sync callback has
 been completed after calling wl_display_dispatch()

Any events queued on the wl_display main queue are dispatched by
wl_display_dispatch(), but it is uncertain whether the frame sync callback
for the previous wl_surface_commit() has been completed after this function
call returns.
This patch makes sure to wait for the frame sync callback for every
frame.
---
 ext/wayland/gstwaylandsink.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 817b2cf..10d9218 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -800,9 +800,8 @@ gst_wayland_sink_render (GstBaseSink * bsink, GstBuffer * buffer)
 
   meta = gst_buffer_get_wl_meta (buffer);
 
-  if (window->redraw_pending) {
+  while (window->redraw_pending)
     wl_display_dispatch (display->display);
-  }
 
   if (meta && meta->sink == sink) {
     GST_LOG_OBJECT (sink, "buffer %p from our pool, writing directly", buffer);
-- 
1.8.0

